code/datums/mind.dm

```diff
@@ -93,8 +93,9 @@
 	if(new_character.mind)		//remove any mind currently in our new body's mind variable
 		new_character.mind.current = null
 
-	if(current.client)
-		current.client.destroy_UI()
+	if (current) // check if current exists otherwise we cannot read current.client
```

Comment:
As I recall, you can combine the first IF and below IF and it will compile and react as you intend for this fix.